### PR TITLE
az webapp new for webapps - support for Linux with Node only (currently)

### PR DESCRIPTION
**Pull request by panchagnula**
Fixing style check errors & fixing receursive zip

Fixing the path the zip needs to created in

Updating DryRun strign to be a json object

dryrun json update

Adding support for detecting runtime from package.json & picking runtime
to create based on supported values

Adding checks for RG & ASP

Bug fixes

FW:

Version ++ & updating history

Style checks

Style check fixes

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [ ] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

**Comment by promptws** [comment]
View a preview at https://prompt.ws/r/Azure/azure-cli/5427
This is an **experimental preview** for @microsoft users.

**Comment by yugangw-msft** [comment]
a few initial questions:
1. Based on old discussion, should not such opinionated commands gets submitted to https://github.com/Azure/azure-cli-extensions?
2. Are we going to repeat the similar commands for all other stacks? Why only Node-8.0?
3. I am not so sure the `new` is the name we can go with

**Comment by ahmedelnably** [comment]
@yugangw-msft for your comment
1. I will let @panchagnula comment on that
2. Other runtimes will come later, we are starting with node for now
3. What is the issue with `new` verb?

**Comment by yugangw-msft** [comment]
> What is the issue with new verb?

It is against the CLI convention that commands which `new up a thing` should be named as `create`; otherwise, people have no clear clue to differentiate between the existing `create` and the new command called `new`.
Once put it in the extension, I would suggest name like `quick-create`, `create-nodejs-web`, etc.
In short, be explicit to avoid confusion.

**Comment by panchagnula** [comment]
@yugangw-msft  Regarding #1 above i guess you are referring to the email with the Advocates, you FWDed me. 

The reasoning is - though the idea of this stemmed from the same conversation - the purpose of this was to give the webapp customers a quick start experience, something we have in the portal. We are starting this with Linux & Node for now (node|8 being the default version). We will follow up with other languages soon - while we decide what would be the next popular language we want to add support for. Hope this clarifies the questions.

Regarding the verb new - i need @ahmedelnably  & Nitasha's opinion as well but we picked new because our quick start does a create as well as deploy the contents using zip deploy.

**Comment by ahmedelnably** [comment]
@yugangw-msft as this is a quick "start" command, would the following work?

az webapp start ....

As we really want to this to be the start of the developer experience with App Service

**Comment by yugangw-msft** [comment]
> @yugangw-msft as this is a quick "start" command, would the following work?
>  az webapp start ....

@ahmedelnably, unfortunately, the `az webapp start` is an existing command which has been there from the V1
```bash
(env) C:\SDK\azure-powershell>az webapp start -h

Command
    az webapp start: Start a web app.

Arguments
    --slot -s          : The name of the slot. Default to the productions slot if not specified.
```

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -468,6 +468,15 @@
     short-summary: Open a web app in a browser.
 """
 
+helps['webapp new'] = """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: There is no precedent for a "new" command in the CLI.  This should definitely be called something different.

File: azure-cli2017.pyproj [hunk 2]
```diff
@@ -14,8 +14,7 @@
     <LaunchProvider>Standard Python launcher</LaunchProvider>
     <InterpreterId>MSBuild|{54f4b6dc-0859-46dc-99bb-b275c9d0aca3}|$(MSBuildProjectFullPath)</InterpreterId>
     <EnableNativeCodeDebugging>False</EnableNativeCodeDebugging>
-    <CommandLineArguments>
-    </CommandLineArguments>
+    <CommandLineArguments>webapp new -n test --dryrun</CommandLineArguments>
```
Comment (tjprescott) [inline azure-cli2017.pyproj]: This needs to be reverted.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py (hunk 1 above)
Comment (panchagnula) [reply src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: yes we will change this once we decide on a verb that also is CLI approved

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/create_util.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    # TODO: call the list_runtimes once there is an API that returs the supported versions
+    if (trunc_version == 4.5 or trunc_version == 4.4 or trunc_version == 6.2 or
+            trunc_version == 6.6 or trunc_version == 6.9 or trunc_version == 6.10 or
+            trunc_version == 6.11 or trunc_version == 8.0 or trunc_version == 8.1):
+        return node_version
+    return version_val
+
+
+def create_resource_group(cmd, rg_name, location):
+    rcf = _resource_client_factory(cmd.cli_ctx)
```
Comment (panchagnula) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/create_util.py]: Change this to copy locally

**Comment by panchagnula** [comment]
this has been moved to extensions project

**Pull request closed without merging**